modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -3202,6 +2919,134 @@ void ONNXImporter::parseCumSum(LayerParams& layerParams, const opencv_onnx::Node
     addLayer(layerParams, node_proto);
 }
 
+// "Equal" "Greater" "Less" "Pow" "Add" "Sub" "Mul" "Div" "Sum" "Min" "Max"
+void ONNXImporter::parseElementWise(LayerParams& layerParams, const opencv_onnx::NodeProto& node_proto_)
+{
+    opencv_onnx::NodeProto node_proto = node_proto_;
+    String op_type = toLowerCase(node_proto.op_type());
+    bool isDiv = op_type == "div";
+
+    layerParams.type = "NaryEltwise";
+    layerParams.set("operation", toLowerCase(node_proto.op_type()));
+
+    // element-wise layers that can have >=1 inputs but actually have one input
+    if (node_proto.input_size() == 1 && (op_type == "max" || op_type == "min" || op_type == "mean" || op_type == "sum"))
+    {
+        layerParams.type = "Identity";
+        addLayer(layerParams, node_proto);
+        return;
+    }
+    else
+    {
+        CV_Assert(node_proto.input_size() == 2);
```

Comment:
I think this is misplaced - NaryEltwiseLayer should support inputs >= 2

Reply:
It should, but following lines are analyzing only two inputs. Practically speaking, inputs >=2 is rare. So I think we can focus on broadcasting in this pull request.

Reply:
1. I think we can remove `else {` and the closing bracket.
2. Would something like this work(not tested)?:

```cpp

size_t consts = 0;
for (size_t i = 0; i < node_proto.input_size(); ++i)
{
    if (layer_id.find(node_proto.input(i)) == layer_id.end())
    {
        ++consts;
    }
}

if (consts == node_proto.input_size())
{
    std::vector<Mat> inputs, output;
    for (size_t i = 0; i < node_proto.input_size(); ++i)
    {
        inputs.push_back(getBlob(node_proto, i));
    }
    runLayer(layerParams, inputs, output);
    CV_Assert(output.size() == 1);
    addConstant(node_proto.output(0), output[0]);
    return;
}
else if (consts > 0)
{
    for (size_t i = 0; i < node_proto.input_size(); ++i)
    {
        if (layer_id.find(node_proto.input(i)) == layer_id.end())
        {
            Mat inp = getBlob(node_proto, i);
            // for cases like a tensor of shape (2,), it will be loaded as shape (2, 1) in OpenCV Mat,
            // but for correct broadcast, we need to make it of shape (1, 2)
            if (constBlobsExtraInfo.find(node_proto.input(i)) != constBlobsExtraInfo.end())
                pre_broadcast_transform(inp, getBlobExtraInfo(node_proto, i).real_ndims);

            // carry the constant by adding a Const node
            LayerParams constParams;
            constParams.name = node_proto.input(i);
            constParams.type = "Const";
            constParams.blobs.push_back(inp);

            opencv_onnx::NodeProto proto;
            proto.add_output(constParams.name);
            addLayer(constParams, proto);
        }
    }
}

```

Reply:
Tested on my side with 2 inputs case and 3 inputs case and worked great. Thanks for the code.

By the way, should we add 3 inputs test case for NaryEltwise in test_layers?

Reply:
Great, thank you! I don't think it's necessary at this point to be honest. If you disagree, I can create a PR to opencv_extra, no problem (so that our CI can properly merge and test everything)